open-metadata-implementation/access-services/asset-lineage/asset-lineage-server/src/main/java/org/odpi/openmetadata/accessservices/assetlineage/handlers/ContextHandler.java

```diff
@@ -312,41 +301,45 @@ private Term buildTerm(EntityDetail glossaryTerm) {
+    private EntityDetail getThePairEntity(String userId, String entityDetailGUID, Relationship relationship) throws org.odpi.openmetadata.frameworks.connectors.ffdc.InvalidParameterException, PropertyServerException, org.odpi.openmetadata.frameworks.connectors.ffdc.UserNotAuthorizedException {
+        String methodname = "getThePairEntity";
         if (relationship.getEntityOneProxy().getGUID().equals(entityDetailGUID)) {
-            return metadataCollection.getEntityDetail(userId, relationship.getEntityTwoProxy().getGUID());
+            EntityDetail entityDetail = repositoryHandler.getEntityByGUID(userId, relationship.getEntityTwoProxy().getGUID(), guidParameter, "Any entity type", methodname);
+            return entityDetail;
         } else {
-            return metadataCollection.getEntityDetail(userId, relationship.getEntityOneProxy().getGUID());
+            EntityDetail entityDetail = repositoryHandler.getEntityByGUID(userId, relationship.getEntityOneProxy().getGUID(), guidParameter, "Any entity type", methodname);
+            return entityDetail;
         }
     }
 
     private List<Relationship> getRelationshipsByAssetId(String userId, String entityId, String relationshipType) throws RepositoryErrorException, UserNotAuthorizedException, EntityNotKnownException, FunctionNotSupportedException, InvalidParameterException, PropertyErrorException, TypeErrorException, PagingErrorException, TypeDefNotKnownException {
 
-        return getRelationshipByType(metadataCollectionForSearch,
+        return getRelationshipByType(
                 userId,
                 entityId,
                 relationshipType);
     }
 
 
-    private List<Relationship> getRelationshipByType(OMRSMetadataCollection metadataCollection, String userId, String entityGUID, String relationshipType) throws UserNotAuthorizedException, EntityNotKnownException, FunctionNotSupportedException, InvalidParameterException, RepositoryErrorException, PropertyErrorException, TypeErrorException, PagingErrorException, TypeDefNotKnownException {
-        String typeGuid = getTypeName(userId, relationshipType, metadataCollection);
+    private List<Relationship> getRelationshipByType(String userId, String entityGUID, String relationshipType) throws UserNotAuthorizedException, EntityNotKnownException, FunctionNotSupportedException, InvalidParameterException, RepositoryErrorException, PropertyErrorException, TypeErrorException, PagingErrorException, TypeDefNotKnownException {
+        String typeGuid = getTypeName(userId, relationshipType);
 
         List<InstanceStatus> instanceStatuses = new ArrayList<>(1);
         instanceStatuses.add(InstanceStatus.ACTIVE);
 
-        List<Relationship> relationshipsForEntity = metadataCollection.getRelationshipsForEntity(userId,
+        List<Relationship> relationshipsForEntity = repositoryHandler.getMetadataCollection().getRelationshipsForEntity(
```

Comment:
@mandy-chessell  I have added a getMetadataCollection() to the RepositoryHandler class, because I could not find an equivalent for getRelationshipsForEntity() within RepositoryHandler itself.